source/application/AMPIC887DriverServer/GCS2Commands/AMGCS2InitializeControllerStateCommand.cpp

```diff
@@ -0,0 +1,346 @@
+#include "AMGCS2InitializeControllerStateCommand.h"
+
+#include "AMGCS2GetMovingStatusCommand.h"
+#include "AMGCS2GetSystemVelocityCommand.h"
+#include "AMGCS2GetCycleTimeCommand.h"
+#include "AMGCS2GetServoModeCommand.h"
+#include "AMGCS2GetCurrentPositionCommand.h"
+#include "AMGCS2GetReferenceResultCommand.h"
+#include "AMGCS2GetLowSoftLimitsCommand.h"
+#include "AMGCS2GetHighSoftLimitsCommand.h"
+#include "AMGCS2GetSoftLimitsStatusCommand.h"
+#include "AMGCS2GetLimitSwitchStatusCommand.h"
+#include "AMGCS2GetPositionUnitsCommand.h"
+#include "AMGCS2GetStepSizeCommand.h"
+#include "AMGCS2GetMinCommandablePositionCommand.h"
+#include "AMGCS2GetMaxCommandablePositionCommand.h"
+#include "AMGCS2GetPivotPointCommand.h"
+#include "AMGCS2GetRecordTriggerSourceCommand.h"
+#include "AMGCS2GetRecordConfigurationCommand.h"
+#include "AMGCS2GetCommandLevelCommand.h"
+#include "AMGCS2GetAvailableParametersCommand.h"
+#include "AMGCS2GetRecorderOptionsCommand.h"
+#include "AMGCS2GetDeviceIdentificationCommand.h"
+
+
+AMGCS2InitializeControllerStateCommand::AMGCS2InitializeControllerStateCommand(AMPIC887ControllerState* controllerState)
+{
+	controllerState_ = controllerState;
+}
+
+AMGCS2InitializeControllerStateCommand::~AMGCS2InitializeControllerStateCommand()
+{
+
+}
+
+bool AMGCS2InitializeControllerStateCommand::validateArguments()
+{
+
+	if(!controllerState_) {
+		lastError_ = "Could not initialize state of controller. No controller state provided";
+		return false;
+	}
+
+	AMGCS2GetMovingStatusCommand movingStatusCommand;
+
+	movingStatusCommand.setController(controller_);
+	movingStatusCommand.run();
+
+	if(movingStatusCommand.runningState() != Succeeded) {
+		lastError_ = "Could not determin motion status of the controller";
+		return false;
+	}
+
+	QFlags<AMGCS2::AxisMovementStatus> axisMovementStatuses =
+			movingStatusCommand.movementStatuses();
+
+	if(axisMovementStatuses) {
+
+		lastError_ = "Cannot load data for a controller while its hexapod is in motion";
+		return false;
+	}
+
+	return true;
+}
+
+
+bool AMGCS2InitializeControllerStateCommand::runImplementation()
```

Comment:
Is this the method that checks to make sure that the hexapod initializes correctly (the comment X,000 lines of code ago)....